sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/filter.hpp

```diff
@@ -132,6 +132,8 @@ class Filter : public rclcpp::Node
     const std::string & filter_name = "pointcloud_preprocessor_filter",
     const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
 
+  ~Filter() override;
```

Comment:
Sorry, maybe I am just not understanding enough,  but should not be a definition for this declaration ? or default or similar?

Reply:
Sorry, this change may not have solved the underlying problem.
The problem is described in following message.
```
/opt/ros/humble/include/message_filters/message_filters/subscriber.h:111:3: error: all paths through this function will call itself [clang-diagnostic-infinite-recursion]
  {
  ^
/opt/ros/humble/include/message_filters/message_filters/subscriber.h:205:3: note: in instantiation of member function 'message_filters::SubscriberBase<>::subscribe' requested here
  ~Subscriber()
  ^
/home/hisaki/workspace/tmp/autoware/install/autoware_pointcloud_preprocessor/include/autoware_pointcloud_preprocessor/autoware/pointcloud_preprocessor/filter.hpp:104:7: note: in instantiation of member function 'message_filters::Subscriber<sensor_msgs::msg::PointCloud2_<std::allocator<void>>>::~Subscriber' requested here
class Filter : public rclcpp::Node
      ^
```
I will reconsider.

Reply:
I realized that this error was due to the implementation of message filter, so I changed pointcloud_preprocessor's cmake to include it as a system library.